actions/ci_connection/action.yaml

```diff
@@ -19,22 +19,75 @@ inputs:
 runs:
   using: "composite"
   steps:
-  - name: Print halt conditions
+  - name: Get current labels
+    shell: bash
+    id: get-labels
+    env:
+      FALLBACK_LABELS: ${{ toJSON(github.event.pull_request.labels) }}
+    run: |
+      # Fetch labels using GitHub API, since labels from context may be stale:
+      # https://github.com/orgs/community/discussions/39062
+      response=$(curl -sL \
+        -H "X-GitHub-Api-Version:2022-11-28" \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer ${{ github.token }}" \
```

Comment:
Can we do without the bearer, it should be fine in post contexts but i would hate for this to somehow leak a write credential (even though noone using this system should be using those)

Reply:
Done (although Actions do redact the actual token in such use cases)